### PR TITLE
Update DEMessageEaseSymbols.kt

### PR DESCRIPTION
**Pull request by bcm314**
Made it as close as possible to original MessagEase
- added tab-key
- added up-/down-key
- some optical changes

### CODE REVIEW

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt [hunk 1]
```diff
@@ -15,6 +29,132 @@ import com.dessalines.thumbkey.utils.KeyboardC
 import com.dessalines.thumbkey.utils.KeyboardDefinition
 import com.dessalines.thumbkey.utils.KeyboardDefinitionModes
 import com.dessalines.thumbkey.utils.SwipeDirection
+import com.dessalines.thumbkey.utils.SwipeNWay
+
+val KB_DE_MESSAGEEASE_SYMBOLS_NUMERIC_KEY_ITEM =
```
Comment (dessalines) [inline app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:34]: Please don't use a custom numeric key, or you will lose any functionality we add later on.

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt [hunk 2]
```diff
@@ -15,6 +29,132 @@ import com.dessalines.thumbkey.utils.KeyboardC
...
+                    KeyC(
+                        display = KeyDisplay.IconDisplay(Icons.Outlined.LinearScale),
+                        action = KeyAction.SwitchPosition,
+                        color = ColorVariant.MUTED,
+                    ),
+            ),
+        backgroundColor = ColorVariant.SURFACE_VARIANT,
+    )
+
+val KB_DE_MESSAGEEASE_SYMBOLS_RETURN_KEY_ITEM =
```
Comment (dessalines) [inline app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:138]: Please don't do a custom return key, use the common one. Or any additions and you will lose all functionality.

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt [hunk 3]
```diff
@@ -15,6 +29,132 @@ import com.dessalines.thumbkey.utils.KeyboardC
...
+                    KeyC(
+                        display = KeyDisplay.IconDisplay(Icons.Outlined.ContentPaste),
+                        action = KeyAction.Paste,
+                        color = ColorVariant.MUTED,
+                    ),
+            ),
+        backgroundColor = ColorVariant.SURFACE_VARIANT,
+    )
+
+val KB_DE_MESSAGEEASE_SYMBOLS_EMOJI_KEY_ITEM =
```
Comment (dessalines) [inline app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:86]: Same here, you will lose all functionality later on. Use the common emoji key.

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt [hunk 4]
```diff
@@ -914,18 +1102,416 @@ val KB_DE_MESSAGEEASE_SYMBOLS_SHIFTED =
             ),
             listOf(
                 SPACEBAR_KEY_ITEM,
-                RETURN_KEY_ITEM,
+                KB_DE_MESSAGEEASE_SYMBOLS_RETURN_KEY_ITEM,
+            ),
+        ),
+    )
+
+val KB_DE_MESSAGEEASE_SYMBOLS_NUMERIC_KEYBOARD =
```
Comment (dessalines) [inline app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:1110]: Get rid of this, we have a common numeric messageease keyboard now in `app/src/main/java/com/dessalines/thumbkey/keyboards/NumericMessagEase.kt`

**Review by dessalines — CHANGES_REQUESTED** [review]
When it comes to our special keys, please just use the existing common ones. Or you will lose all functionality when we add things to them later on.

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt (hunk 1 above)
Comment (bcm314) [reply app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:34]: I've tried to make it as close to MessagEase as possible.

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt (hunk 2 above)
Comment (bcm314) [reply app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:138]: I've tried to make it as close to MessagEase as possible.

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt (hunk 3 above)
Comment (bcm314) [reply app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:86]: I've tried to make it as close to MessagEase as possible.

Comment (dessalines) [reply app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:86]: This isn't messageeease, we have our own custom keys.

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt (hunk 4 above)
Comment (dessalines) [reply app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:1110]: No probs. Apparently messageease has custom numeric keyboards for each language, but they should live in their own files, IE like `app/src/main/java/com/dessalines/thumbkey/keyboards/ENMessagEaseNumeric.kt`

File: app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt (hunk 3 above)
Comment (bcm314) [reply app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:86]: > This isn't messageeease, we have our own custom keys.

Indeed, you're right.

Maybe I should be even open for new keyboard-layouts - they could have advantages. Or create my own, best fitting my conditions (I'm writing with one finger, not thumb). But better to have a standard, not learning new layouts (and difficulty to forget old ones) when changing the app, even not the optimum.

Comment (dessalines) [reply app/src/main/java/com/dessalines/thumbkey/keyboards/DEMessageEaseSymbols.kt:86]: No probs! Ya if you wanted to make your own layout also, I'd be glad to add it.

**Comment by dessalines** [comment]
I can re-open if you want to pick this up again.

**Pull request closed without merging**